pkg/tools/doc/types.go

```diff
@@ -0,0 +1,318 @@
+type SwaggerV2Spec struct {
+	Definitions map[string]*KclOpenAPIType `json:"definitions"`
+	Paths       map[string]interface{}     `json:"paths"`
+	Swagger     string                     `json:"swagger"`
+	Info        SpecInfo                   `json:"info"`
+}
+
+// SpecInfo defines KCL package info
+type SpecInfo struct {
+	Title       string `json:"title"`
+	Version     string `json:"version"`
+	Description string `json:"description"`
+}
+
+// KclOpenAPIType defines the KCL representation of SchemaObject field in Swagger v2.0.
+// And the mapping between kcl type and the Kcl OpenAPI type is:
+//
+// ## Basic Types
+//
+// | KCL Type   |  KCL OpenAPI Type (format) |
+//
+// | ---------- | -------------------------- |
+//
+// |    str     |        string              |
+//
+// |    int     |       integer(int64)       |
+//
+// |   float    |       number(float)        |
+//
+// |   bool     |       bool                 |
+//
+// ## Composite Types
+//
+// |  KCL Types    |     KCL OpenAPI Types         |
+//
+// |  ----------   |   --------------------------  |
+//
+// |     list      |  type:array, items: itemType  |
+//
+// |     dict      | type: object, additionalProperties: valueType, x-kcl-dict-key-type: keyType |
+//
+// |     union     | type: object, x-kcl-union-types: unionTypes                   |
+//
+// |    schema     | type: object, properties: propertyTypes, required, x-kcl-type |
+//
+// | nested schema | type: object, ref: jsonRefPath |
+type KclOpenAPIType struct {
+	Type                 SwaggerTypeName            // object, string, array, integer, number, bool
+	Format               TypeFormat                 // type format
+	Default              string                     // default value
+	Enum                 []string                   // enum values
+	ReadOnly             bool                       // readonly
+	Description          string                     // description
+	Properties           map[string]*KclOpenAPIType // schema properties
+	Required             []string                   // list of required schema property names
+	Items                *KclOpenAPIType            // list item type
+	AdditionalProperties *KclOpenAPIType            // dict value type
+	Example              string                     // example
+	ExternalDocs         string                     // externalDocs
+	KclExtensions        *KclExtensions             // x-kcl- extensions
+	Ref                  string                     // reference to schema path
+}
+
+// SwaggerTypeName defines possible values of "type" field in Swagger v2.0 spec
+type SwaggerTypeName string
+
+const (
+	Object  SwaggerTypeName = "object"
+	String  SwaggerTypeName = "string"
+	Array   SwaggerTypeName = "array"
+	Integer SwaggerTypeName = "integer"
+	Number  SwaggerTypeName = "number"
+	Bool    SwaggerTypeName = "bool"
+)
+
+// KclTypeName defines the KCL representation of basic types
+type KclTypeName string
+
+const (
+	Str      KclTypeName = "str"
```

Comment:
U can use types at the `tools/gen` package.

Reply:
fixed